diff --git a/node_modules/react-native-gesture-handler/src/components/Swipeable.tsx b/node_modules/react-native-gesture-handler/src/components/Swipeable.tsx
index c654c66..f832a95 100644
--- a/node_modules/react-native-gesture-handler/src/components/Swipeable.tsx
+++ b/node_modules/react-native-gesture-handler/src/components/Swipeable.tsx
@@ -89,6 +89,8 @@ export interface SwipeableProps
    */
   overshootFriction?: number;
   
+  onSwipeStart?: () => void;
+
   /**
    * Called when left action panel gets open.
    */
@@ -301,6 +303,9 @@ export default class Swipeable extends Component<
   private onHandlerStateChange = (
     ev: HandlerStateChangeEvent<PanGestureHandlerEventPayload>
   ) => {
+    if (ev.nativeEvent.state === State.ACTIVE) {
+      this.props.onSwipeStart && this.props.onSwipeStart()
+    }
     if (ev.nativeEvent.oldState === State.ACTIVE) {
       this.handleRelease(ev);
     }
