diff --git a/node_modules/react-native-bootsplash/ReactNativeBootsplash.res b/node_modules/react-native-bootsplash/ReactNativeBootsplash.res
index b12a8d6..56bbc92 100644
--- a/node_modules/react-native-bootsplash/ReactNativeBootsplash.res
+++ b/node_modules/react-native-bootsplash/ReactNativeBootsplash.res
@@ -1,10 +1,10 @@
 type config = {fade: bool}
 
 @bs.module("react-native-bootsplash") @bs.scope("default")
-external hide: config => unit = "hide"
+external hide: config => Js.Promise.t<unit> = "hide"
 
 @bs.module("react-native-bootsplash") @bs.scope("default")
-external show: config => unit = "show"
+external show: config => Js.Promise.t<unit> = "show"
 
 // Note that this binding requires BuckleScript >= 8.2.0
 type visibilityStatus = [#visible | #hidden | #transitioning]
@@ -12,10 +12,22 @@ type visibilityStatus = [#visible | #hidden | #transitioning]
 @bs.module("react-native-bootsplash") @bs.scope("default")
 external getVisibilityStatus: unit => Js.Promise.t<visibilityStatus> = "getVisibilityStatus"
 /*
- ## Usage
+## Usage
 
- ```re
- ReactNativeBootSplash.hide({fade: true});
- ReactNativeBootSplash.show({fade: false});
- ```
- */
+```rescript
+ ReactNativeBootsplash.hide({fade: true})->ignore
+ ReactNativeBootSplash.show({fade: false})->ignore
+```
+
+Or
+
+ ```rescript
+ReactNativeBootsplash.hide({fade: true})->Js.Promise.then_(() => {
+  Js.log("RN BootSplash: fading is over")
+  Js.Promise.resolve()
+}, _)->Js.Promise.catch(error => {
+  Js.log(("RN BootSplash: cannot hide splash", error))
+  Js.Promise.resolve()
+}, _)->ignore
+```
+*/
